### ISSUE TITLE
Do we need to catch depsgraph?

### ISSUE THREAD

**Issue by Durman** [nortikin/sverchok#4609]
I faced with bug which seems relevant to Geometry Nodes. I was not able to reproduce this problems with the Subdivision modifier. It appears when undo is applied to GN modifier and Get Objects node is in Post mode.

https://user-images.githubusercontent.com/28003269/182774414-708f7905-343a-45a1-9975-d83520c211ba.mp4

On this line Blender crashes.

https://github.com/nortikin/sverchok/blob/70cd8f1df51414b3b81c8a60c7048a8ffd3a4573/nodes/scene/get_objects_data.py#L327

Most likely report to Blender tracker should be made. I don't think we misuse Blender API here.

**Comment by Durman** [comment]
Probably it's not a problem of Blender. I'm able reproduce the crash if I'm using fresh depsgraph before the problem line.

```py
sv_depsgraph=bpy.context.evaluated_depsgraph_get()
obj = sv_depsgraph.objects[obj.name]
obj_data = obj.to_mesh(preserve_all_data_layers=True, depsgraph=sv_depsgraph)
```

I assume this function is unable to give really fresh depsgraph

https://github.com/nortikin/sverchok/blob/70cd8f1df51414b3b81c8a60c7048a8ffd3a4573/core/handlers.py#L23-L33

**Comment by zeffii** [comment]
maybe we need to use the _undo-handler_ as a reasonable point in runtime to re-evaluate the depsgraph. `undo_pre` or `undo_post` ?

**Comment by Durman** [comment]
Yes, this might solve the problem, but...

I'm curious why we have such a twisted way to get depsgrath? It seems is design to call `evaluated_depsgraph_get` as less a possible. So it based on an assumption that each call of the method is expensive but is it? Probably it's a common knowledge because I did not saw any discussion about that here #2511 (where it was initially introduced) or here #2852.

In the documentation we can find that calling this can be expensive indeed. But also there is a condition. 

> This might be expensive **if changes were made made to the scene**,

Since Sverchok usually does not change scene data the call should not be expensive then. And if Sverchok does changes something then depsgrapth should probably be reevaluated any way?

And finally. I made some speed tests with this scene.

![image](https://user-images.githubusercontent.com/28003269/182831652-52766879-c031-41c9-b58c-54a4b6c59e29.png)

It's quite heavy. it's weight is about 500mb. I ran this scrip several times.

```py
from timeit import timeit
print(timeit('bpy.context.evaluated_depsgraph_get()', 'import bpy', number=1))
```

First call took this `0.000571` all next calls took about this `8.000002708286047e-06` and this is even after when I moved, removed, added, edited objects.

So my conclusion is that there is no need in centralization of access to the depsgraph.

**Comment by zeffii** [comment]
the introduction of the use of depsgraph for `to_mesh` introduced a problem which was solved and declared resolved as soon as we got it to work. Calling the function at the time was also triggering a nodetree update (iirc), so reducing the calls was seen as helpful.

If calling `bpy.context.evaluated_depsgraph_get()` is an operation that is "cached" already by blender and doesn't retrigger a nodetree-update, then i agree if you are saying we can just call it when needed.  -- not sure the `sv_get_depsgraph` internals are needed the.
